Tests/UnitTests/EdgeBridgeTests.swift

```diff
@@ -1443,9 +1466,55 @@ class EdgeBridgeTests: XCTestCase, AnyCodableAsserts {
         assertAppIDInDispatchedEvent("")
     }
 
+    // MARK: Customer Perspective tests
+
+    func testHandleTrackEvent_withActiveApplicationState_dispatchesEdgeRequestEvent_withCustomerPerspectiveForeground() {
+        var fakeHelper = FakeEdgeBridgeHelper()
+        fakeHelper.applicationState = .active
+        edgeBridge.bridgeHelper = fakeHelper
+
+        assertCPInDispatchedEvent("foreground")
+    }
+
+    func testHandleTrackEvent_withBackgroundApplicationState_dispatchesEdgeRequestEvent_withCustomerPerspectiveBackground() {
+        var fakeHelper = FakeEdgeBridgeHelper()
+        fakeHelper.applicationState = .background
+        edgeBridge.bridgeHelper = fakeHelper
+
+        assertCPInDispatchedEvent("background")
+    }
+
+    func testHandleTrackEvent_withInactiveApplicationState_dispatchesEdgeRequestEvent_withCustomerPerspectiveForeground() {
+        var fakeHelper = FakeEdgeBridgeHelper()
+        fakeHelper.applicationState = .inactive
+        edgeBridge.bridgeHelper = fakeHelper
+
+        assertCPInDispatchedEvent("foreground")
+    }
+
+    func testHandleTrackEvent_withNilApplicationState_dispatchesEdgeRequestEvent_withCustomerPerspectiveForeground() {
+        var fakeHelper = FakeEdgeBridgeHelper()
+        fakeHelper.applicationState = nil
+        edgeBridge.bridgeHelper = fakeHelper
+
+        assertCPInDispatchedEvent("foreground")
```

Comment:
little too many layers of abstraction here - can we use assertInDispatchedEvent(appId: applicationIdentifier, customerPerspective: customerPerspective) instead?